score.cpp

```diff
@@ -28,31 +29,64 @@ Score::Score(std::string specialPath){
     try{
-        std::filesystem::path pathToRun = std::filesystem::current_path();
-        //std::cout << "Path to project: " << pathToProject << std::endl;
+        std::filesystem::path DefaultFilePath = getDefaultFilePath();
+        std::filesystem::path pathToFloder = DefaultFilePath / "2048_data";
+        std::filesystem::create_directories(pathToFloder);
+        pathToStorage = pathToFloder / "scores.txt";
+        std::cout << "Path to score storage: " << pathToStorage << std::endl;
+
+        scoreStorage.open(pathToStorage, std::ios::app);
+        if(!scoreStorage.is_open()){
+            std::cerr << "File not open and score not saved" << std::endl;
+            return;
+        }
+        std::cout << "File is open..." << std::endl;
+        scoreStorage << 0;
+        scoreStorage.close();
+        std::cout << "File is closed" << std::endl;
 
-        while(pathToRun != pathToRun.root_path()){ //Looking for project directory (it will be mistake if CMakeLists.txt is deleted)
-            std::cout << "Current path: " << pathToRun << std::endl;
-            if(std::filesystem::exists(pathToRun / "CMakeLists.txt")){
+    //     std::filesystem::path pathToRun = std::filesystem::current_path();
+    //     //std::cout << "Path to project: " << pathToProject << std::endl;
 
-                std::filesystem::path pathToProject = pathToRun;
-                std::cout << "Path to project: " << pathToProject << std::endl;
+    //     while(pathToRun != pathToRun.root_path()){ //Looking for project directory (it will be mistake if CMakeLists.txt is deleted)
+    //         std::cout << "Current path: " << pathToRun << std::endl;
+    //         if(std::filesystem::exists(pathToRun / "CMakeLists.txt")){
 
-                std::filesystem::path pathToFloder = pathToProject / "data";
-                std::filesystem::create_directories(pathToFloder);
+    //             std::filesystem::path pathToProject = pathToRun;
+    //             std::cout << "Path to project: " << pathToProject << std::endl;
 
-                pathToStorage = pathToFloder / "scores.txt";
-                std::cout << "Path to score storage: " << pathToStorage << std::endl;
-                break;
-            }
-            pathToRun = pathToRun.parent_path();
-        }
+    //             std::filesystem::path pathToFloder = pathToProject / "data";
+    //             std::filesystem::create_directories(pathToFloder);
+
+    //             pathToStorage = pathToFloder / "scores.txt";
+    //             std::cout << "Path to score storage: " << pathToStorage << std::endl;
+    //             break;
+    //         }
+    //         pathToRun = pathToRun.parent_path();
+    //     }
     }
     catch(const std::filesystem::filesystem_error &err){
         std::cerr << "File processing error: " << err.what() << std::endl;
     }
 }
 
+std::string Score::getDefaultFilePath(){
+    const char* homeDir;
+
+    #ifdef _WIN32
+        homeDir = getenv("USERPROFFILE");
```

Comment:
Have you tried to use something cross-platform from std::filesystem or Qt to achieve this?

Reply:
done